disnake/state.py

```diff
@@ -284,19 +284,7 @@ def clear(
         self, *, views: bool = True, application_commands: bool = True, modals: bool = True
     ) -> None:
         self.user: ClientUser = MISSING
-        # Originally, this code used WeakValueDictionary to maintain references to the
-        # global user mapping.
-
-        # However, profiling showed that this came with two cons:
-
-        # 1. The __weakref__ slot caused a non-trivial increase in memory
-        # 2. The performance of the mapping caused store_user to be a bottleneck.
```

Comment:
probably keep some form of this and mention that even with these drawbacks we are continuing to use this system

Reply:
https://github.com/DisnakeDev/disnake/pull/858/commits/f3e48b4851d9e180820d495b9ea12ebc6bfdd89e